ldap-client.go

```diff
@@ -438,6 +456,27 @@ func (lc *LDAPClient) ChangeOpenLDAPUserPassword(username, oldPassword, newPassw
 	return nil
 }
 
+func (lc *LDAPClient) GetUserByCN(userCN,uidAttr string)(uid string, err error){
+	searchRequest := ldap.NewSearchRequest(
+		userCN,
+		ldap.ScopeWholeSubtree, ldap.NeverDerefAliases, 0, 0, false,
+		"(objectClass=*)",
+		[]string{"cn", uidAttr},
+		nil,
+	)
+
+	sr, err := lc.Conn.Search(searchRequest)
+	if err != nil {
+		return "", err
+	}
+	if len(sr.Entries) != 1 {
+		return "", nil
```

Comment:
IIUC, it means that we found more than one user, right? 
If yes, and assuming it is an illegal behavior, you should return an error.